### PR TITLE
[feature] create a module for IAM role with the policies integration

### PR DESCRIPTION
**Pull request by Dingying0410**
### Summary
**Motivation**:
When creating a new IAM role, we typically need to specify the following resources in TF
-`aws_iam_policy_document` specifying which entities can assume that role 
- the `aws_iam_role` with the `assume_role_policy` as `aws_iam_policy_document`
- and uausually with the `aws_iam_role_policy` and `aws_iam_role_policy_attachment` specifying the policies granted to the roles

This logic has existed in many places of the repo, (and if I am not missing something obvious, it seemed I haven't found one module available in this repo that can specify a general `iam-role` for this purpose), so I created this module which can simulate the process of creating a new IAM role like in the AWS console. 

### Test Plan
- the unit test (wip, as I can't test creating the role locally) 
- LP's repp PR 648

### References
LP's repp PR 648

### CODE REVIEW

File: aws-iam-role-policies/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+output role_name {
```
Comment (cziie) [inline aws-iam-role-policies/outputs.tf:1]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`role_name` output has no description

File: aws-iam-role-policies/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  value = aws_iam_role.role.id
+}
+
+output role_arn {
```
Comment (cziie) [inline aws-iam-role-policies/outputs.tf:5]: **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
`role_arn` output has no description

File: aws-iam-role-policies/module_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	env := tftest.UniqueID()
+	service := tftest.UniqueID()
+	owner := tftest.UniqueID()
+	defer tftest.DeleteRole(t, roleName) //nolint
+
+	terraformOptions := tftest.Options(
+		tftest.IAMRegion,
+		map[string]interface{}{
+			"role_name": roleName,
+			"project": project,
```
Comment (cziie) [inline aws-iam-role-policies/module_test.go:21]: **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
File is not `gofmt`-ed with `-s` (gofmt)

File: aws-iam-role-policies/variables.tf [hunk 4]
```diff
@@ -0,0 +1,48 @@
+variable project {
```
Comment (ryanking) [inline aws-iam-role-policies/variables.tf:1]: Please, add a variable for `iam_path` (and use it on all `iam` resources).

**Review by ryanking — CHANGES_REQUESTED** [review]
This is great and will be very useful. Some comments left inline, but also I think we should rename to `aws-iam-role`.

File: aws-iam-role-policies/variables.tf (hunk 4 above)
Comment (Dingying0410) [reply aws-iam-role-policies/variables.tf]: added, and we can only use that for the IAM role in this module

File: aws-iam-role-policies/variables.tf [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+variable owner {
+  type        = string
+  description = "Owner for tagging and naming. See [doc](../README.md#consistent-tagging)"
+}
+
+variable principals {
+  type        = list(object({ type = string, identifiers = list(string) }))
+  description = "The Principal element in the role's trust policy to specify who can assume the role, this will be used in the `assume_role_policy` for the IAM role."
+  default     = []
```
[the remark this answers is not in the document]
Comment (Dingying0410) [reply aws-iam-role-policies/variables.tf]: yeah I changed `principals` to a required variable

**Review by ryanking — CHANGES_REQUESTED** [review]

File: aws-iam-role/module_test.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/chanzuckerberg/go-misc/tftest"
+)
+
+func TestAWSIAMROle(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ryanking) [reply aws-iam-role/module_test.go:9]: nevermind I will fix this

**Review by ryanking — APPROVED** [review]

**Pull request merged**